peps/pep-0780.rst

```diff
@@ -0,0 +1,469 @@
+PEP: 780
+Title: ABI features as environment markers
+Author: Klaus Zimmermann <klaus_zimmermann@gmx.de>,
+        Ralf Gommers <ralf.gommers@gmail.com>
+Sponsor: Lysandros Nikolaou <lisandrosnik@gmail.com>
+Discussions-To: https://discuss.python.org/t/60007
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 21-Mar-2025
+Python-Version: 3.14
+
+..
+    If adopted, the contents of this pep should be integrated into the
+    canonical documentation in the Python Packaging User Guide and the
+    following directive should be added here:
+    .. canonical-pypa-spec:: :ref:`packaging:dependency-specifiers`
+
+Abstract
+========
+
+This PEP defines new environment markers for project dependencies.
+:pep:`508` (later moved to :ref:`packaging:dependency-specifiers`) introduced
+environment markers to specify dependencies based on rules that describe
+when the dependency should be used.
+This PEP extends the environment markers to allow specifying dependencies
+based on specific ABI features of the Python interpreter.
+For this, it defines a set of `ABI Features`_ and specifies how they are made
+available via an `Addition to the Python Standard Library`_ in the form of a
+new attribute ``sys.abi_features``, as well as for `Environment Markers`_ as a
+new marker variable, ``sys_abi_features``.
+
+Motivation
+==========
+
+:pep:`508` established environment markers to specify dependencies based on
+environment conditions. Since its adoption in 2015, the Python ecosystem has
+evolved and new features warrant their own environment markers.
+
+The primary motivation for extending the allowed set of environment markers now
+is free-threaded CPython [#python-free-threading]_ emerging. Currently there is
```

Comment:
I think this could be tightened up -- this PEP only proposes one new environment marker for ABI features. Perhaps (though you may want to rephrase):

```suggestion
In 2015, :pep:`508` established environment markers to specify dependencies
based on environment conditions.
The development of free-threaded CPython [#python-free-threading]_
has underlined the need for an environment marker to discriminate between
different ABI features that the interpreter was built with.
For example, currently there is
```